onnxruntime/core/framework/session_state.cc

```diff
@@ -1065,7 +1113,8 @@ Status SessionState::FinalizeSessionStateImpl(const std::basic_string<PATH_CHAR_
       session_options.GetConfigOrDefault(kOrtSessionOptionsConfigDisablePrepacking, "0");
 
   if (disable_prepacking != "1") {
-    ORT_RETURN_IF_ERROR(PrepackConstantInitializedTensors(constant_initializers_use_count));
+    // TODO: Remove const_cast ugliness
+    ORT_RETURN_IF_ERROR(PrepackConstantInitializedTensors(constant_initializers_use_count, const_cast<SessionOptions&>(session_options)));
```

Comment:
can we avoid const_cast?

Reply:
const_cast goes away with changed public interface